### PR TITLE
Group 4 Solution

### PR DESCRIPTION
**Pull request by cupcakehunter**
Elliot and Akshith

### CODE REVIEW

File: tic_tac_toe/game.py [hunk 1]
```diff
@@ -24,7 +26,10 @@ def _position_is_valid(position):
 
     Returns True if given position is valid, False otherwise.
     """
-    pass
+    if not isinstance(position, tuple) or len(position) != 2 or not isinstance(position[0], int) or not isinstance(position[1], int):
+        return False
+        
+    return 0 <= position[0] <= 2 and 0 <= position[1] <= 2
```
Comment (balmf) [inline tic_tac_toe/game.py:32]: Checking that the elements in the position tuple are ints is a very good idea, this is something we did not think to do. If I were to nit-pick I suppose the strings '0', '1', '2'... could also be allowed also.
I also like how you've written the check that the ints are valid within the return statement without the use of a separate if statement.

File: tic_tac_toe/game.py [hunk 2]
```diff
@@ -63,22 +74,31 @@ def _check_winning_combinations(board, player):
...
     """
-    pass
+    for i in range(3):
+        if _is_winning_combination(board, ((i,0), (i,1), (i,2)), player) or _is_winning_combination(board, ((0,i), (1,i), (2,i)), player):
+            return player
+    
+    if _is_winning_combination(board, ((0,0), (1,1), (2,2)), player) or _is_winning_combination(board, ((0,2), (1,1), (2,0)), player):
+        return player
+        
+    return None
```
Comment (balmf) [inline tic_tac_toe/game.py:84]: Good use of for loop rather than hardcoding every winning combination and checking against one by one.

**Comment by balmf** [comment]
All in all can't see any problems. Nice readable, clean and consistent code.

File: tic_tac_toe/game.py (hunk 1 above)
Comment (jon003) [inline tic_tac_toe/game.py:32]: Nice pythonic way to write this!

**Pull request closed without merging**